test/e2e/network/topology_hints.go

```diff
@@ -143,15 +172,9 @@ var _ = common.SIGDescribe("Feature:Topology Hints", func() {
 		}
 
 		ginkgo.By("keeping requests in the same zone")
-		for i, nodeName := range nodeNames {
-			// Iterate through max of 3 nodes
-			if i > 2 {
-				break
-			}
-			fromZone, ok := nodesByZone[nodeName]
-			if !ok {
-				framework.Failf("Expected zone to be specified for %s node", nodeName)
-			}
+		for fromZone, nodes := range zonesWithNodes {
+			nodeName := nodes[0]
```

Comment:
I didn't check in detail, do we guarantee that we always have at least one node per zone? otherwise this will panic

Reply:
I don't think panic will happen. `zonesWithNodes` is set on L161. Each zone should map to a list with at least one Node.

Reply:
If we move to `map[string]string{}` as I suggested above, I think this will be a non issue.

Reply:
done